pkg/store/sqlite/transactions.go

```diff
@@ -168,6 +168,39 @@ func (s *Store) DeleteTransaction(ctx context.Context, id uuid.UUID) (err error)
 	return tx.Commit()
 }
 
+const (
+	latestSecEnvSQL            = "SELECT * FROM secure_envelopes WHERE id=:envelopeID ORDER BY timestamp DESC LIMIT 1"
+	latestSecEnvByDirectionSQL = "SELECT * FROM secure_envelopes WHERE id=:envelopeID AND direction=:direction ORDER BY timestamp DESC LIMIT 1"
+)
+
+func (s *Store) LatestSecureEnvelope(ctx context.Context, envelopeID uuid.UUID, direction string) (env *models.SecureEnvelope, err error) {
+	var tx *sql.Tx
+	if tx, err = s.BeginTx(ctx, &sql.TxOptions{ReadOnly: true}); err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	var result *sql.Row
+	if direction == "" || direction == models.DirectionAny {
```

Comment:
Pretty sure my brain mixed up direction and source after I responded to your question, but it's all clear again.